addons/purchase_stock/models/purchase.py

```diff
@@ -37,15 +39,15 @@ def _default_picking_type(self):
                  'order_line.move_ids.state',
                  'order_line.move_ids.picking_id')
     def _compute_picking(self):
+        orders = defaultdict(lambda: set())
+        move_ids = self.env['stock.move'].search([('purchase_line_id.order_id', 'in', self.ids)]).ids
+        move_ids.extend(self.env['stock.move'].search([('origin_returned_move_id', 'in', move_ids)]).ids)
```

Comment:
Again == `self.mapped("order_line.move_ids.returned_move_ids")`. (maybe use a temporary variable with result of `self.mapped("order_line.move_ids")` ?